### PR TITLE
check rule on issue

### PR DESCRIPTION
**Pull request by SimonLab**
see #58 
This PR:
- check for breaking rule on issue for label "in-progress" with no assignees
- wait 30s and check the status of the issue again
- Add a comment on the issue to notify the user

### CODE REVIEW

**Comment by SimonLab** [comment]
I have deploy this PR on heroku to test it, https://dwylbot.herokuapp.com so you can play with it too. If you create an event on an issue (new issue, change labels, new assignee...) dwylbot will add a comment after 30s to let you know that you need to correct the issue (it it's wrong)

File: config/config.exs [hunk 1]
```diff
@@ -36,6 +36,9 @@ config :ueberauth, Ueberauth.Strategy.Github.OAuth,
   client_id: System.get_env("GITHUB_CLIENT_ID"),
   client_secret: System.get_env("GITHUB_CLIENT_SECRET")
 
+# define name of the github application
+config :dwylbot, :github_app_name, "simonlabapp"
```
Comment (SimonLab) [inline config/config.exs]: Sorry I need to add the name of the application in an environment variable

File: test/controllers/event_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+defmodule Dwylbot.EventTestController do
+  use Dwylbot.ConnCase
+  alias Poison.Parser, as: PP
+
+  test "POST /event/new - no error", %{conn: conn} do
+    payload = "./test/fixtures/add_label.json"
+    |> File.read!()
+    |> PP.parse!()
```
Comment (samhstn) [inline test/controllers/event_controller_test.exs:8]: We've been leaving off the `()`s here

File: test/fixtures/add_label.json [hunk 3]
```diff
@@ -0,0 +1,172 @@
+{
```
Comment (samhstn) [inline test/fixtures/add_label.json:1]: Maybe reference the curl command you ran to get this

You could for instance add it at the top of the JSON file as:
{ "curl_command": "curl https://githu...",...}, but I'm not sure if it's _that_ good practice, but I'd personally add it

I'm not sure if this is correct, but my actual preference is to have the absolute minimum in my fixtures to make the test pass and add more to the fixture as you need it

File: web/controllers/rules/rules.ex [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    in_progress = Enum.any?(labels, fn(l) -> l["name"] == "in-progress" end)
+    if in_progress && Enum.empty?(assignees) do
+      [%{error_type: "inprogress_noassignees", author: author}]
+    else
+      []
+    end
+  end
+
+  def compare(errors, check) do
+    Enum.filter check, fn(e) -> has_error?(errors, e) end
```
Comment (samhstn) [inline web/controllers/rules/rules.ex:17]: Nice 👍
I'm a fan of this lack of bracket syntax

File: web/controllers/rules/rules.ex [hunk 5]
```diff
@@ -0,0 +1,33 @@
+defmodule Dwylbot.Rules do
+  @moduledoc """
+  Functions to check the wrokflow contribution rules
+  """
+  def check(payload, author) do
```
Comment (samhstn) [inline web/controllers/rules/rules.ex:5]: Given this is a pure function, could you add a `@doc` test to show how it works

File: test/controllers/event_controller_test.exs (hunk 2 above)
Comment (nelsonic) [reply test/controllers/event_controller_test.exs:8]: @shouston3 is the omission of parens for function invocation documented somewhere in one of our repos?

File: test/fixtures/add_label.json (hunk 3 above)
Comment (nelsonic) [reply test/fixtures/add_label.json:1]: @SimonLab this could be a good enhancement in the next iteration.

**Review by nelsonic — APPROVED** [review]
@SimonLab code looks great! 😍 
@shouston3 thanks for peer-reviewing. 🎉

File: test/controllers/event_controller_test.exs (hunk 2 above)
Comment (SimonLab) [reply test/controllers/event_controller_test.exs:8]: I've got a warning with I omit the brackets: ```warning: parentheses are required when piping into a function call.```
Maybe it's fine if there are not parameters?

Comment (samhstn) [reply test/controllers/event_controller_test.exs:8]: parens are needed with params, but without params I prefer to not have them in

Comment (samhstn) [reply test/controllers/event_controller_test.exs:8]: @nelsonic @SimonLab Preferences about this are documented in [this elixir styleguide](https://github.com/christopheradams/elixir_style_guide#pipe-operator)

I've opened an issue for documenting omission of parens (and other syntax decisions) in dwyl/learn-elixir [here](https://github.com/dwyl/learn-elixir/issues/47)

**Pull request merged**